### ISSUE TITLE
Failure with PNG and space optimizations.

### ISSUE THREAD

**Issue by evoskuil** [libbitcoin/libbitcoin#715]
There is a bug in the PNG library when using space optimization levels, such as `CXXFLAGS=-Os` which will show up in a test run failure. Should check back with libpng at some point.

**Comment by thecodefactory** [comment]
Note: This needs to be re-tested after libbitcoin#891 was merged.

**Comment by thecodefactory** [comment]
@evoskuil @fpelliccioni

I was able to reproduce this before the fix:

```
 cat test.log 
Running 1933 test cases...
Platform: linux
Compiler: GNU C++ version 4.9.2
STL     : GNU libstdc++ version 20141220
Boost   : 1.57.0
unknown location(0): fatal error in "png__size_one__success": memory access violation at address: 0x040003e7: no mapping at fault address
test/utility/endian.cpp(165): last checkpoint
```

I was also able to confirm that the fix in libbitcoin#891 does in fact resolve this (at least on GNU/Linux using gcc).

@pmienk IIRC, there was a reason we wanted to use "-Os" for some benefit on the travis builds.  If that's the case, I think we're all clear now to try again.
